### ISSUE TITLE
INVITE/SHARE: Sharing UX issues for public and private topics

### ISSUE THREAD

**Issue by BeccaMelhuish** [citizenos/citizenos-fe#1250]
Trying to get a share link for this public topic, I found I couldn't:
(In live) 

![image](https://github.com/citizenos/citizenos-fe/assets/61320526/5d7ab368-1009-4f36-aebf-f8dcf742ee5e)

**Comment by ilmartyrk** [comment]
@BeccaMelhuish  @kevincrepin I believe just copy pasting the topic link is OK UX as the share/join link would add the new user to the topic and they might not want to. Also join links might have other permission levels so we cannot display it to users who don't have the permissions to see it.

**Comment by BeccaMelhuish** [comment]
I think copying the URL might be OK, as a couple of users have shown in tests that they would naturally do this if wanting to share the topic (though not in this context exactly - actually as admins of private topics, but more on that at the end). Two thoughts though, on what could be preferable:

1. @kevincrepin do you think we should have a 'Share' button somewhere for participants, with a pop-up with a button that allows them to click to 'Copy URL', or this is unnecessary as they could just copy paste it themselves? Am thinking it's more the expected UX, but unless we have the usual options 'Share via WhatsApp' 'Share via email' etc, it doesn't actually serve much purpose? (Though having those options could be good, actually, depending how complicated this would be for @ilmartyrk? But this is just an assumption based on the standard UX you see everywhere else, no users have yet asked for this. And I guess this is more a 'nice to have' for future). 

2. @ilmartyrk what currently happens if a user copies the URL to a private topic or group? As if we have that as the UX for sharing public ones (and as users are already showing that they naturally do this), then we should also make it work for private ones. Is it this 'Not found' screen below that they get? Or an 'Insufficient permissions' notification?

![image](https://github.com/user-attachments/assets/7790d07b-d181-4f8c-aa18-2e09b33c3f94)

@kevincrepin I'm thinking that the best UX would be to follow Google docs example? So have a share button available to everyone, but if you share it with someone who hasn't been added then they get a page that:

- As a quick fix in the short term, tells them "Sorry, you don't have access to this topic. Contact the topic admin to request an invite". (Obviously they might not know who that is, but in a lot of cases if it's a private topic, they're probably able to find this out - i.e. by asking whoever sent them the link). And with 'topic' switched to 'group' when it's a group. 

- But ideally, when there is time for it, should we have a page similar to the 'Request access' page for google doc? So when they land, and have logged in, it tells them something like: "Sorry, you don't have access to this topic." With an option below to "Request access" (maybe with which level of rights they'd like to request?). And an indication of which email they are logged in with as well, in case they only don't have access for that reason.  

Also, some user insights that show a need for the latter, in a couple of tests, admins have said/shown that they would just copy the URL of their topic in order to share it with participants, in a private topic. So if we had the Google docs style 'Request access' screen, this would at least mean that if an admin did use this approach, their participants would be able to request access rather than just coming up against a 'Not found' or whatever message they would currently get.

**Comment by ilmartyrk** [comment]
Quick fix is to create the suggested 404 page for topics and maybe for groups too. And in the later phase we will implement the request access flow, but this will be inside another issue scope.

**Comment by BeccaMelhuish** [comment]
@kevincrepin Please provide a design for a page saying the below :) 

"Sorry, you don't have access to this topic. Contact the topic admin to request an invite"

@ilmartyrk If it's simple, we could also include this in the quick fix?

"You are currently logged in with: X (email address)?"

**Comment by kevincrepin** [comment]
Ehm, why don't have the invite & share button for participants? Is it only for admins now? I always imagined it would be available for everyone (and why not?)...
So **yes** to the invite and share button. Can work exactly the same as the one the admin gets, just with no role setting option.

And for topic that user doesn't have access to: 
![Screenshot 2024-09-27 at 15 06 33](https://github.com/user-attachments/assets/47fdbd59-2054-49c2-a4a7-554e3f4f780a)

@ilmartyrk : https://www.figma.com/design/gOJ4i00538e0qaP6IKIPVE/Platform-2.0---Design?node-id=14755-143526&t=3hHy3U6W9ohEF2Lu-1

**Comment by BeccaMelhuish** [comment]
@kevincrepin I think for private topics it makes sense that only admin(s) can choose who has access (to give them control over access in case the content is sensitive), but good yes let's add the invite and share button for everyone in public topics then :) 

Request for access looks great! Only 2 thoughts: 
- Could we have a note at the bottom (I guess above or instead of the 'go to dashboard') saying "You are currently logged in with email@example.com". Then this should hopefully prompt them to realise if their lack of access is simply due to being logged in with the wrong account. 
- When they click on the 'Request access' button, will they have a pop-up in which they choose which level of rights they are requesting? Could we have a design for that too? :) 

For @ilmartyrk when back, as I guess the 'request access' system is a bigger bit of work to set-up, as a quick fix for the short term when could just use the design above but with the text:

**Sorry, you don't have access to this topic** 
Try contacting the topic admin to request an invite. 
You are currently logged in with email@example.com

**Comment by Lidziya-p** [comment]
@BeccaMelhuish https://www.figma.com/design/ODWxKEHuAt3DGkrJwRZJ8a/Platform-2.5---Design?node-id=14755-143526&t=MjrBZH5ZZxBptZQf-1 Like this?

**Comment by BeccaMelhuish** [comment]
Wonderful yes, thank you @Lidziya-p!

Copy-wise for these two highlighted lines though I'd suggest: 

- **Request access**
- **Add a reason for your request (optional)** 

![image](https://github.com/user-attachments/assets/338037e8-78c8-445e-abce-fa9e6667d38e)

I'm thinking now that we will also need to think about the UX of what happens next (admin gets sent a request, how do they approve or change the rights but I will make a separate issue for that one. 

**So @ilmartyrk to summarise for this issue (the 'quick fix') it's just to implement this part (without the 'Request access' button):**

![image](https://github.com/user-attachments/assets/322d8551-fcac-4363-a8da-5955b07647a5)

**Comment by BeccaMelhuish** [comment]
**Dev task summary:** 

- [ ] Add the invite and share button in **public** topics for participants and editors (as it already is for admins). 
- [ ] Design should be similar to the top part from [#1751](https://github.com/citizenos/citizenos-fe/issues/1751) but with the following differences: 
- 'Share link / QR'
- '**Invite others via a join link**' (Just the heading no need for any text underneath) 
- no 'renew link' button
- Middle sentence removed from QR code explanation
![Image](https://github.com/user-attachments/assets/c511018c-3048-457c-94c1-7c737fa58d5f)

- [x] Make sure if the admin creates a new participants rights link, it changes in this window for participants as well.

- [ ] 'Invite participants' tab can be the same as it currently is, but make sure it is 'participants' not 'members' as per [#1563](https://github.com/citizenos/citizenos-fe/issues/1563). (Please close that one once fixed).

**Comment by ilmartyrk** [comment]
@BeccaMelhuish the task list a little confusing. Initial part I can understand, basically for public topic participants to have the ability to share the participant level join link to other people. Private topic item I don't understand what you mean by that. If a join link is clicked a user will get to join the topic if the link itself is valid. Or is the last view meant for if the topic link is sent directly not the join link. Or is it meant to be for expired invites?

Other tasks I estimate to be 4h

**Comment by BeccaMelhuish** [comment]
@ilmartyrk yeah it's the "meant for if the topic link is sent directly not the join link" one :) 

So if you get sent the URL from the browser rather than a proper join link in a private topic to which you're not added you currently just get this: 

![Image](https://github.com/user-attachments/assets/fcbf2eb7-68d2-445b-8071-142bfc87d0b5)

So it would be to replace that. But that's a good point re expired invite, I'm not sure what the user gets currently if the invite has expired. But that's probably a separate issue :)

**Comment by BeccaMelhuish** [comment]
Actually @ilmartyrk I will split the last part out into a separate issue as it can be lower priority, will remove it from the to-do list now :)

**Comment by BeccaMelhuish** [comment]
Just noting that this caused issues with the Democra-Si case, where a link was shared to a private topic which was not a join link
